sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue

```diff
@@ -127,48 +137,28 @@ export default defineComponent({
 
       if (
         applicationProgressDetails.value.scholasticStandingChangeType ===
-        StudentScholasticStandingChangeType.StudentDidNotCompleteProgram
+          StudentScholasticStandingChangeType.StudentDidNotCompleteProgram ||
+        applicationProgressDetails.value.pirStatus ===
+          ProgramInfoStatus.declined ||
+        applicationProgressDetails.value.exceptionStatus ===
+          ApplicationExceptionStatus.Declined ||
+        applicationProgressDetails.value.firstCOEStatus ===
+          COEStatus.declined ||
+        applicationProgressDetails.value.secondCOEStatus === COEStatus.declined
       ) {
-        // Application is complete has an error.
-        applicationEndStatus.value = {
-          endStatusType: "error",
-          endStatusIcon: "fa:fas fa-exclamation-circle",
-        };
+        // One of the requests or confirmations is declined.
+        statusIconDetails.value = new StatusIconDetails("error");
       } else if (
         applicationProgressDetails.value.appealStatus ===
         StudentAppealStatus.Pending
       ) {
         // Application is complete but has warnings.
-        applicationEndStatus.value = {
-          endStatusType: "warning",
-          endStatusIcon: "fa:fas fa-exclamation-triangle",
-        };
+        statusIconDetails.value = new StatusIconDetails("warning");
       } else if (
         // Application is complete.
-        applicationProgressDetails.value.firstCOEStatus ===
-          COEStatus.completed &&
-        (!applicationProgressDetails.value.secondCOEStatus ||
-          applicationProgressDetails.value.secondCOEStatus ===
-            COEStatus.completed)
+        applicationProgressDetails.value.firstCOEStatus === COEStatus.completed
       ) {
-        applicationEndStatus.value = {
-          endStatusType: "success",
-          endStatusIcon: "fa:fas fa-check-circle",
-        };
-      } else if (
-        // One of the requests or confirmations is declined.
-        applicationProgressDetails.value.pirStatus ===
-          ProgramInfoStatus.declined ||
-        applicationProgressDetails.value.exceptionStatus ===
-          ApplicationExceptionStatus.Declined ||
-        applicationProgressDetails.value.firstCOEStatus ===
-          COEStatus.declined ||
-        applicationProgressDetails.value.secondCOEStatus === COEStatus.declined
-      ) {
-        applicationEndStatus.value = {
-          endStatusType: "error",
-          endStatusIcon: "fa:fas fa-exclamation-circle",
-        };
+        statusIconDetails.value = new StatusIconDetails("success");
```

Comment:
Yes, I thought the same, but after our last Team conversation I thought that it would potentially cause more questions than benefits and that is why I did a refactor here. Please let me know if it is still ok.